src/openssl/ssl/server.cr

```diff
@@ -60,18 +63,24 @@ class OpenSSL::SSL::Server
   #
   # This method calls `@wrapped.accept` and wraps the resulting IO in a SSL socket (`OpenSSL::SSL::Socket::Server`) with `context` configuration.
   def accept : OpenSSL::SSL::Socket::Server
-    OpenSSL::SSL::Socket::Server.new(@wrapped.accept, @context, sync_close: @sync_close)
+    new_server(@wrapped.accept)
   end
 
   # Implements `::Socket::Server#accept?`.
   #
   # This method calls `@wrapped.accept?` and wraps the resulting IO in a SSL socket (`OpenSSL::SSL::Socket::Server`) with `context` configuration.
   def accept? : OpenSSL::SSL::Socket::Server?
     if socket = @wrapped.accept?
-      OpenSSL::SSL::Socket::Server.new(socket, @context, sync_close: @sync_close)
+      new_server(socket)
     end
   end
 
+  private def new_server(io)
+    server = OpenSSL::SSL::Socket::Server.new(io, @context, sync_close: @sync_close)
+    server.accept if @start_immediately
```

Comment:
```suggestion
    server.accept if start_immediately?
```

Reply:
This is not needed. A call is a bit more expensive for the compiler in terms of compiling.

Reply:
Put another way: both do the same thing and we don't prefer one over the other.

Reply:
I'd argue that in the vein of OOP we should prefer using getters over ivars (ability to overload 'em by subclasses)

Reply:
I then call YAGNI